arlo-client/src/components/Audit/Setup/Settings/index.tsx

```diff
@@ -1,24 +1,149 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { toast } from 'react-toastify'
+import { useParams } from 'react-router-dom'
+import { Formik, FormikProps, Form, Field, ErrorMessage } from 'formik'
+import { RadioGroup, Radio } from '@blueprintjs/core'
 import { IAudit } from '../../../../types'
 import FormButtonBar from '../../../Form/FormButtonBar'
 import FormButton from '../../../Form/FormButton'
 import { ISidebarMenuItem } from '../../../Atoms/Sidebar'
+import { IValues } from './types'
+import FormWrapper from '../../../Form/FormWrapper'
+import FormSection from '../../../Form/FormSection'
+import { Select } from '../../EstimateSampleSize'
+import { generateOptions, ErrorLabel } from '../../../Form/_helpers'
+import { parse as parseNumber } from '../../../../utils/number-schema'
+import FormField from '../../../Form/FormField'
+import schema from './schema'
+import useAuditSettings from '../useAuditSettings'
 
 interface IProps {
   audit: IAudit
   nextStage: ISidebarMenuItem
   prevStage: ISidebarMenuItem
 }
 
-const Settings: React.FC<IProps> = ({ nextStage, prevStage }: IProps) => {
+const Settings: React.FC<IProps> = ({
+  nextStage,
+  prevStage,
+  audit,
+}: IProps) => {
+  const { electionId } = useParams()
+  const [isLoading, setIsLoading] = useState(false)
+  const [
+    { electionName = '', randomSeed = '', riskLimit = 10, online = true },
+    updateState,
+  ] = useAuditSettings(electionId!)
+  const submit = async (values: IValues) => {
+    try {
+      setIsLoading(true)
+      const response = updateState({
+        ...values,
+        riskLimit: parseNumber(values.riskLimit), // Formik stringifies internally
+      })
+      if (!response) return
+      nextStage.activate()
+    } catch (err) {
+      toast.error(err.message)
+    }
+  }
   return (
-    <div>
-      <p>Audit Settings</p>
-      <FormButtonBar>
-        <FormButton onClick={prevStage.activate}>Back</FormButton>
-        <FormButton onClick={nextStage.activate}>Next</FormButton>
-      </FormButtonBar>
-    </div>
+    <Formik
+      initialValues={{
+        electionName,
+        randomSeed,
+        riskLimit,
+        online,
+      }}
+      validationSchema={schema}
+      onSubmit={submit}
+      enableReinitialize
+    >
+      {({ handleSubmit, setFieldValue, values }: FormikProps<IValues>) => (
+        <Form data-testid="form-one">
+          <FormWrapper title="Audit Settings">
+            <FormSection>
+              {/* eslint-disable jsx-a11y/label-has-associated-control */}
+              <label htmlFor="audit-name" id="audit-name-label">
+                Election Name
+                <Field
+                  id="election-name"
+                  aria-labelledby="election-name-label"
+                  name="electionName"
+                  disabled={!!audit.frozenAt}
```

Comment:
Where is `frozenAt` being set? I know that @benadida had originally built an API to "freeze" the audit, but I took that out of the API proposal I wrote. In the world I proposed, an audit's settings should no longer be updated as soon as the first round starts (which you can tell when you hit the /election/<election_id>/round API), but there's no indicator in the election settings API.

Reply:
@jonahkagan Oh, I missed that distinction. Hm. So, to clarify, what you're saying is that when the 'Launch Audit' button is hit and launching is confirmed on the last step of the setup flow, that starts the first round? And having rounds is exactly identical to setup being completed and locked?

If so, I'll need to add another state management hook to keep track of the rounds api responses. Right now, I'm just using the regular audit /status endpoint.

Reply:
Yes and yes to your questions. You understand me correctly.

I mentally bucketed `/audit/status` as a single-juris flow endpoint. It seems a bit tricky to try to make it return correct info for both single-juris and multi-juris flows. So better not to rely on it here.

Is hitting the rounds API a good way to manage the overall control flow? I figured the rounds are the real world "control flow" of the audit, so it makes sense. But if there's another API that would work better, let me know and let's talk through it.

Reply:
@jonahkagan So would I be pinging `GET /election/<election_id>/round` then? What response will I be expecting pre-launch vs post-launch?

It's a shift from how the front end architecture was handling things before. On the single-page flow the /status endpoint was the single source of truth for all data. It's pinged at every interaction with the server (plus some more times) to update everything. Now, everything is broken up into a bunch of endpoints that appear to work more or less independently of each other, and there is no longer a single way to access the single source of truth. So I'm just trying to figure out which ones need to be pinged when and for which part of the app.

What I'm seeing for this particular piece, since whether or not all of the setup forms are locked or not is a pretty critical piece of information, I'll end up having to ping an endpoint (whether it's the rounds endpoint or something else) on every rerender to make sure the data is up-to-date. So as long as it's light-weight it doesn't particularly matter which endpoint it is, I don't think.

Reply:
Regardless, for this PR, I'll need to leave the frozen bit as is for now, since that's a more involved change that needs its own PR.

Reply:
Yes, you'd hit `GET /election/<election_id>/round`. Pre-launch, you'd get this `{"rounds": []}`. Post-launch, you'd get `{"rounds": [ ... array of round objects, ordered by roundNum ... ]}`.

Currently, this endpoint simply reads the rounds from the db. It's a two line function, so I'd say pretty lightweight! 😄 

Makes me think we should also put in some server-side logic to block requests to all the set up endpoints after the first round is created.